Formula/u/uffizzi.rb

```diff
@@ -0,0 +1,80 @@
+class Uffizzi < Formula
+  desc "Self-serve developer platforms in minutes, not months with k8s virtual clusters"
+  homepage "https://uffizzi.com"
+  url "https://github.com/UffizziCloud/uffizzi_cli/archive/refs/tags/v2.0.35.tar.gz"
+  sha256 "95a541b167307aac7a69416dd621134eefae4f910722fb14d87da0b0b5b116bf"
+  license "Apache-2.0"
+
+  uses_from_macos "ruby"
+
+  resource "activesupport" do
+    url "https://rubygems.org/gems/activesupport-6.1.4.1.gem"
+    sha256 "44b781877c2189aa15ca5451e2d310dcedfd16c01df1106f68a91b82990cfda5"
+  end
+
+  resource "awesome_print" do
+    url "https://rubygems.org/gems/awesome_print-1.9.2.gem"
+    sha256 "e99b32b704acff16d768b3468680793ced40bfdc4537eb07e06a4be11133786e"
+  end
+
+  resource "faker" do
+    url "https://rubygems.org/gems/faker-2.22.0.gem"
+    sha256 "6525d450cf5bb98ddeeb8bdd235bd2323f4fab829f491855397fb79f72ba48d9"
+  end
+
+  resource "launchy" do
+    url "https://rubygems.org/gems/launchy-2.5.2.gem"
+    sha256 "8aa0441655aec5514008e1d04892c2de3ba57bd337afb984568da091121a241b"
+  end
+
+  resource "minitar" do
+    url "https://rubygems.org/gems/minitar-0.9.gem"
+    sha256 "23c0bebead35dbfe9e24088dc436c8a233d03f51d365a686b9a11dd30dc2d588"
+  end
+
+  resource "securerandom" do
+    url "https://rubygems.org/gems/securerandom-0.2.2.gem"
+    sha256 "5fcb3b8aa050bac5de93a5e22b69483856f70d43affeb883bce0c58d71360131"
+  end
+
+  resource "sentry-ruby" do
+    url "https://rubygems.org/gems/sentry-ruby-5.11.0.gem"
+    sha256 "27f603638d75d28b974def362792f442ae39e3e1c1496427910f9a0f434f3a71"
+  end
+
+  resource "thor" do
+    url "https://rubygems.org/gems/thor-1.2.2.gem"
+    sha256 "2f93c652828cba9fcf4f65f5dc8c306f1a7317e05aad5835a13740122c17f24c"
+  end
+
+  resource "tty-prompt" do
+    url "https://rubygems.org/gems/tty-prompt-0.23.1.gem"
+    sha256 "fcdbce905238993f27eecfdf67597a636bc839d92192f6a0eef22b8166449ec8"
+  end
+
+  resource "tty-spinner" do
+    url "https://rubygems.org/gems/tty-spinner-0.9.3.gem"
+    sha256 "0e036f047b4ffb61f2aa45f5a770ec00b4d04130531558a94bfc5b192b570542"
+  end
+
+  def install
+    ENV["GEM_HOME"] = libexec
+    ENV["GEM_PATH"] = libexec
+
+    resources.each do |r|
+      r.fetch
+      system "gem", "install", r.cached_download, "--ignore-dependencies",
+             "--no-document", "--install-dir", libexec
+    end
+
+    system "gem", "install", "uffizzi-cli", "-v", "2.0.35", "--no-document", "--install-dir", libexec
+
+    bin.install Dir["#{libexec}/bin/*"]
+
+    bin.env_script_all_files(libexec, GEM_HOME: ENV["GEM_HOME"], GEM_PATH: ENV["GEM_PATH"])
+  end
+
+  test do
+    assert_match("2.0.35", shell_output("#{bin}/uffizzi version").strip)
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.